compiler/rustc_passes/src/liveness.rs

```diff
@@ -329,6 +329,48 @@ impl<'tcx> Visitor<'tcx> for IrMaps<'tcx> {
             }
         }
 
+        // Allow todo! macro
+        /*
+            Skips checking for unused variables when the trailing expression
+            of the body is a panic with a message that contains "not yet implemented".
```

Comment:
I would still include something saying that this is a somewhat hacky way to check if it's `todo!`.